components/VercelChat/preview-attachment.tsx

```diff
@@ -0,0 +1,56 @@
+import type { Attachment } from 'ai';
+import { Loader2, FileIcon } from 'lucide-react';
+
+export const PreviewAttachment = ({
+  attachment,
+  isUploading = false,
+}: {
+  attachment: Attachment;
+  isUploading?: boolean;
+}) => {
+  const { name, url, contentType } = attachment;
+
+  return (
+    <div data-testid="input-attachment-preview" className="flex flex-col gap-2">
+      <div className="w-20 h-16 aspect-video bg-muted rounded-md relative flex flex-col items-center justify-center overflow-hidden">
+        {contentType ? (
+          contentType.startsWith('image') ? (
+            // Image preview
+            <img
+              key={url}
+              src={url}
+              alt={name ?? 'An image attachment'}
+              className="rounded-md size-full object-cover"
+            />
+          ) : contentType === 'application/pdf' ? (
+            // PDF icon
+            <div className="flex items-center justify-center text-zinc-500">
+              <FileIcon size={24} />
+              <span className="text-xs ml-1">.pdf</span>
+            </div>
+          ) : (
+            // Generic file icon
+            <div className="flex items-center justify-center text-zinc-500">
+              <FileIcon size={24} />
+            </div>
+          )
```

Comment:
only images are supported right?
Why do you have so much extra logic if you're only supporting images?

Reply:
It is a extension, in case in future we other files than just images.

Removing it for now